src/pattern.class.ts

```diff
@@ -141,7 +137,7 @@ export class Pattern {
           this.source.naturalWidth === 0 ||
           this.source.naturalHeight === 0))
     ) {
-      return '';
+      return null;
```

Comment:
doesn't this cause a problem in canvas?
I understand why you prefer this
I would look into usage and do `|| ''`

Reply:
i tried, if you set up a non valid color for canvas, you still get black.
so null or empty string it doesn't change.
empty string is as non valid as null for a fillStyle/strokeStyle.

createPattern api from context can also return null if pattern creation fails.